sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -219,6 +219,10 @@ def process_encoded(self, encoded_windowed_values):
 
   def monitoring_infos(self, transform_id, tag_to_pcollection_id):
     # type: (str, Dict[str, str]) -> Dict[FrozenSet, metrics_pb2.MonitoringInfo]
+
+    # TODO(BEAM-9979): Fix race condition where reused DataInputOperation
```

Comment:
Operators have a `reset` method that could be used here.

Reply:
Thanks, Java SDK harness doesn't have something like this plumbed through to its operators.